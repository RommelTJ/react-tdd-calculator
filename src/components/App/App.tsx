import React from 'react';

import './App.css';
import Calculator from "../Calculator/Calculator";

const App = () => {
  return (
    <div role="presentation" className="app-container">
      <Calculator />
    </div>
  )
};

export default App;
