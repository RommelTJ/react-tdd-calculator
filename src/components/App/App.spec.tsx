import React from 'react';
import { render, RenderResult, screen } from '@testing-library/react'
import App from './App';

describe('App', () => {

  let wrapper: RenderResult;

  beforeEach(() => wrapper = render(<App/>));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a div', () => {
    const divs = screen.getAllByRole('presentation');
    expect(divs).toHaveLength(1);
  });

  it('should render the Calculator Component', () => {
    const calculator = screen.getByTestId("calculator");
    expect(calculator).toBeTruthy();
  });
});
