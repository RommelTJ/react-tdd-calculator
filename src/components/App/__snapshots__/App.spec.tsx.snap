// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app-container"
        role="presentation"
      >
        <div
          class="calculator-container"
          data-testid="calculator"
        >
          <div
            class="display-container"
          >
            <p
              class="display-value"
            >
              0
            </p>
          </div>
          <div
            class="keypad-container"
          >
            <div
              class="numbers-container"
            />
            <div
              class="operators-container"
            />
            <div
              class="submit-container"
            >
              <div
                class="key-container submit-key"
              >
                <p
                  class="key-value"
                >
                  =
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app-container"
      role="presentation"
    >
      <div
        class="calculator-container"
        data-testid="calculator"
      >
        <div
          class="display-container"
        >
          <p
            class="display-value"
          >
            0
          </p>
        </div>
        <div
          class="keypad-container"
        >
          <div
            class="numbers-container"
          />
          <div
            class="operators-container"
          />
          <div
            class="submit-container"
          >
            <div
              class="key-container submit-key"
            >
              <p
                class="key-value"
              >
                =
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
